### PR TITLE
Add create offer change

### PR DESCRIPTION
**Pull request by achilleasa**
This PR introduces a new Change type called `CreateOfferChange`. This new change type will be used for re-creating application offers if defined within the bundle (see https://github.com/juju/charm/pull/277).

### CODE REVIEW

File: changes.go [hunk 1]
```diff
@@ -71,6 +71,7 @@ func FromData(config ChangesConfig) ([]Change, error) {
 			return nil, errors.Trace(err)
 		}
 	}
+	resolver.handleOffers(addedApplications)
```
Comment (SimonRichardson) [inline changes.go]: What do we do about `kubernetes` here, I think it's correct, but we should be sure.

Comment (achilleasa) [reply changes.go]: The function assumes that the bundle is already validated (at least according to the comment above its declaration) so we should probably deal with this when we are doing validations, right?

Comment (SimonRichardson) [reply changes.go]: I believe so.

**Comment by achilleasa** [comment]
$$merge$$

**Comment by mitechie** [comment]
Not sure if it needs :shipit: or $$merge$$ but here's both!

**Comment by achilleasa** [comment]
!!build!!

**Comment by achilleasa** [comment]
$$merge$$

**Pull request merged**